compiler.ts

```diff
@@ -424,10 +427,59 @@ export class Compiler implements CmdVisitor<void>, ExprVisitor<void> {
   }
 }
 
+// Note, the CompilerResult[] will include unmerged warnings...
```

Comment:
I actually like this abstraction quite a bit! Though I'd also like to refactor Compiler so it once has different code paths for Programs versus Cmds. Maybe there can even be a separate ProgramCompiler and BaseCompiler...